### PR TITLE
upgrade bower-json + code clean up

### PR DESCRIPTION
**Pull request by anuccio1**
In order to fix an issue with running `srclib-javascript` in node 10, we are upgrading the version of `bower-json` being used, as well as the following changes:

- remove vendored `node_modules` folder
- modify Makefile to install deps
- remove `depresolve` and `graph` methods

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+10.24.1
```
Comment (taliamccormick) [inline .nvmrc:1]: Why do we want to remove the nvmrc?

File: .gitmodules [hunk 2]
```diff
@@ -1,12 +0,0 @@
-[submodule "testdata/case/javascript-nodejs-sample-0"]
```
Comment (taliamccormick) [inline .gitmodules:1]: Same question here, intentional removal?

File: .nvmrc (hunk 1 above)
Comment (anuccio1) [reply .nvmrc:1]: this is adding one that didn't exist previously

File: .gitmodules (hunk 2 above)
Comment (anuccio1) [reply .gitmodules:1]: yup. this is dead code that isn't used anywhere

File: .nvmrc (hunk 1 above)
Comment (taliamccormick) [reply .nvmrc:1]: Ah, shoot, misread that -- thanks!

**Review by jagonalez — APPROVED** [review]
LGTM

**Pull request merged**